src/KahanSummation.jl

```diff
@@ -29,6 +29,59 @@ else
     promote_sys_size_add(x::T) where {T} = Base.r_promote(+, zero(T)::T)
 end
 
+
+import Base.TwicePrecision
+
+
+function plus_kbn(x::T, y::T) where {T}
```

Comment:
Should these be restricted to `T<:Real`? Also do they necessarily have to be the same type? Using `+` as we do here should take care of any promotion that needs to happen.